pkg/kv/bulk/sst_batcher.go

```diff
@@ -451,20 +450,51 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 		b.ms.LastUpdateNanos = timeutil.Now().UnixNano()
 	}
 
-	if err := b.addSSTable(ctx, start, end, b.sstFile.Data()); err != nil {
-		return err
+	stats := b.ms
+	summary := b.rowCounter.BulkOpSummary
+	data := b.sstFile.Data()
+	batchTS := b.batchTS
+	var reserved int64
+
+	fn := func(ctx context.Context) error {
+		if err := b.addSSTable(ctx, batchTS, start, end, data, stats); err != nil {
+			return err
+		}
+		b.mu.Lock()
+		summary.DataSize += int64(size)
+		b.totalRows.Add(summary)
+		b.stats.batchWait += timeutil.Since(beforeFlush)
+		b.stats.dataSize += size
+		b.mu.Unlock()
+		if reserved != 0 {
+			b.mem.Shrink(ctx, reserved)
+		}
+		return nil
 	}
 
-	b.rowCounter.DataSize += b.sstWriter.DataSize
-	b.totalRows.Add(b.rowCounter.BulkOpSummary)
-	b.stats.batchWait += timeutil.Since(beforeFlush)
-	b.stats.dataSize += sz(b.sstWriter.DataSize)
-	return nil
+	if reason == rangeFlush {
```

Comment:
can we add a comment here similar to what you have in the commit message talking about async AddSSTable requests for range flushes

Reply:
done.